install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml

```
@@ -1,96 +1,87 @@
+    {{- end }}
+  {{- end }}
 spec:
   selector:
     matchLabels:
       k8s-app: cilium
-{{- if .Values.keepDeprecatedLabels }}
+    {{- if .Values.keepDeprecatedLabels }}
       kubernetes.io/cluster-service: "true"
-{{- end }}
-{{- with .Values.updateStrategy }}
+    {{- end }}
+  {{- with .Values.updateStrategy }}
   updateStrategy:
     {{- toYaml . | trim | nindent 4 }}
-{{- end }}
+  {{- end }}
   template:
     metadata:
       annotations:
-{{- if and .Values.prometheus.enabled (not .Values.prometheus.serviceMonitor.enabled) }}
+      {{- if and .Values.prometheus.enabled (not .Values.prometheus.serviceMonitor.enabled) }}
         prometheus.io/port: "{{ .Values.prometheus.port }}"
         prometheus.io/scrape: "true"
-{{- end }}
-{{- if .Values.rollOutCiliumPods }}
+      {{- end }}
+      {{- if .Values.rollOutCiliumPods }}
         # ensure pods roll when configmap updates
         cilium.io/cilium-configmap-checksum: {{ include (print $.Template.BasePath "/cilium-configmap.yaml") . | sha256sum | quote }}
-{{- end }}
+      {{- end }}
         # This annotation plus the CriticalAddonsOnly toleration makes
         # cilium to be a critical pod in the cluster, which ensures cilium
         # gets priority scheduling.
         # https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/
         scheduler.alpha.kubernetes.io/critical-pod: ""
-{{- with .Values.podAnnotations }}
+      {{- with .Values.podAnnotations }}
         {{- toYaml . | nindent 8 }}
-{{- end }}
+      {{- end }}
       labels:
         k8s-app: cilium
-{{- if .Values.keepDeprecatedLabels }}
+      {{- if .Values.keepDeprecatedLabels }}
         kubernetes.io/cluster-service: "true"
-{{- end }}
-{{- with .Values.podLabels }}
+      {{- end }}
+      {{- with .Values.podLabels }}
         {{- toYaml . | nindent 8 }}
-{{- end }}
+      {{- end }}
     spec:
-{{- if .Values.affinity }}
+      {{- if .Values.affinity }}
       affinity:
-{{ toYaml .Values.affinity | indent 8 }}
-{{- end }}
-{{- if .Values.imagePullSecrets }}
+        {{- toYaml .Values.affinity | nindent 8 }}
+      {{- end }}
+      {{- if .Values.imagePullSecrets }}
       imagePullSecrets:
-{{ toYaml .Values.imagePullSecrets | indent 6 }}
-{{- end }}
+        {{- toYaml .Values.imagePullSecrets | nindent 8 }}
+      {{- end }}
       containers:
-{{- if .Values.sleepAfterInit }}
-      - command: [ "/bin/bash", "-c", "--" ]
-        args: [ "while true; do sleep 30; done;" ]
+      - name: cilium-agent
+        image: {{ include "cilium.image" .Values.image | quote }}
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        {{- if .Values.sleepAfterInit }}
+        command:
+        - /bin/bash
+        - -c
+        - --
+        args:
+        - |
+          while true;
+            do sleep 30;
+          done
```

Nit: I'd put the `do` higher to keep just the commands inside the block, but no strong preference.
```suggestion
          while true; do
            sleep 30;
          done
```